pytest_testrail/plugin.py

```diff
@@ -97,34 +101,45 @@ def get_testrail_keys(items):
 
 class PyTestRailPlugin(object):
     def __init__(
-            self, client, assign_user_id, project_id, suite_id, cert_check, tr_name):
+            self, client, assign_user_id, project_id, suite_id, cert_check, tr_name, run_id=0, plan_id=0, version=''):
         self.assign_user_id = assign_user_id
         self.cert_check = cert_check
         self.client = client
         self.project_id = project_id
         self.results = []
         self.suite_id = suite_id
-        self.testrun_id = 0
         self.testrun_name = tr_name
+        self.testrun_id = run_id
+        self.testplan_id = plan_id
+        self.version = version
 
     # pytest hooks
 
     @pytest.hookimpl(trylast=True)
     def pytest_collection_modifyitems(self, session, config, items):
         tr_keys = get_testrail_keys(items)
-        if self.testrun_name is None:
-            self.testrun_name = testrun_name()
-
-        self.create_test_run(
-            self.assign_user_id,
-            self.project_id,
-            self.suite_id,
-            self.testrun_name,
-            tr_keys
-        )
+
+        if self.testplan_id and self.is_testplan_available():
+            print('Use existing testplan "ID={}"'.format(self.testplan_id))
```

Comment:
Why printing directly to console? Couldn't we return this information in the pytest header hook?

Reply:
I am printing this information to allow easier debugging. And that's the way used by the maintainer (I usually prefer to use "logging" module). But I will have a look to "pytest header hook". It seems to be interesting.

Reply:
Pytest header hook appears to be the best way to go, but I'm fine with this for now